dcos-log/api/v2/handlers.go

```diff
@@ -38,75 +38,110 @@ const (
 	eventStreamContentType = "text/event-stream"
 )
 
+type errSetupFilesAPIReader struct {
+	msg  string
+	code int
+}
+
+// Error implements errors interface.
+func (e errSetupFilesAPIReader) Error() string {
+	return e.msg
+}
+
 // TODO: pass a request to logError to enhance logging. (url, query parameters, any relevant headers, and so on)
 // logError is a http.Error wrapper, that also emits an error to a console
 func logError(w http.ResponseWriter, msg string, code int) {
 	http.Error(w, msg, code)
 	logrus.Errorf("%s; http code: %d", msg, code)
 }
 
-func filesAPIHandler(w http.ResponseWriter, req *http.Request) {
+func setupFilesAPIReader(req *http.Request, urlPath string, opts ...reader.Option) (r *reader.ReadManager, err error) {
+
 	cfg, ok := middleware.FromContextConfig(req.Context())
 	if !ok {
-		logError(w, "invalid context, unable to retrieve a config object", http.StatusInternalServerError)
-		return
+		return nil, errSetupFilesAPIReader{
+			msg:  "invalid context, unable to retrieve a config object",
+			code: http.StatusInternalServerError,
+		}
 	}
 
 	client, ok := middleware.FromContextHTTPClient(req.Context())
 	if !ok {
-		logError(w, "invalid context, unable to retrieve an *http.Client object", http.StatusInternalServerError)
-		return
+		return nil, errSetupFilesAPIReader{
+			msg:  "invalid context, unable to retrieve an *http.Client object",
```

Comment:
Here and elsewhere, you might be able to use the `%T` formatting directive here to avoid creating a string literal with the type you were expecting. So something like 

```
msg: fmt.Sprintf("invalid context, unable to retrieve %T object", client)
```

Since `client` in this case is associated with a type.

Reply:
TIL :)